### PR TITLE
Add Marshal SecureString apis

### LINKED ISSUE #13339: Bring the Marshal APIs that use SecureString to netstandard2.0
Once https://github.com/dotnet/coreclr/pull/7985 is merged and a new CoreCLR package available, expose these APIs in the reference assembly for System.Runtime.InteropServices.

```csharp
    public static class Marshal {
        public static IntPtr SecureStringToBSTR(SecureString s);
        public static IntPtr SecureStringToCoTaskMemAnsi(SecureString s);
        public static IntPtr SecureStringToCoTaskMemUnicode(SecureString s);
        public static IntPtr SecureStringToGlobalAllocAnsi(SecureString s);
        public static IntPtr SecureStringToGlobalAllocUnicode(SecureString s);
    }
```

### PR DESCRIPTION
**Pull request by alexperovich**
Fixes #13339

### CODE REVIEW

**Comment by alexperovich** [comment]
@dotnet-bot test Innerloop Linux ARM Emulator SoftFP Release Cross please (emulator failed)

**Comment by alexperovich** [comment]
@dotnet-bot test Innerloop Linux ARM Emulator SoftFP Release Cross Build please (emulator failed)

File: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs [hunk 1]
```diff
@@ -721,6 +721,16 @@ public static partial class Marshal
...
+        public static IntPtr SecureStringToBSTR(System.Security.SecureString s) { throw null; }
+        [System.CLSCompliant(false)]
+        public static IntPtr SecureStringToCoTaskMemAnsi(System.Security.SecureString s) { throw null; }
+        [System.CLSCompliant(false)]
+        public static IntPtr SecureStringToCoTaskMemUnicode(System.Security.SecureString s) { throw null; }
+        [System.CLSCompliant(false)]
+        public static IntPtr SecureStringToGlobalAllocAnsi(System.Security.SecureString s) { throw null; }
+        [System.CLSCompliant(false)]
+        public static IntPtr SecureStringToGlobalAllocUnicode(System.Security.SecureString s) { throw null; }
         [System.ComponentModel.EditorBrowsableAttribute((System.ComponentModel.EditorBrowsableState)(1))]
```
Comment (danmosemsft) [inline src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs]: @tijoytom please open a separate issue for that and mark it api-needs-work or api-ready-for-review as appropriate.

**Comment by stephentoub** [comment]
We should add tests for these (and move the tests from System.Security.SecureString to System.Runtime.InteropServices).  Otherwise, LGTM.

**Comment by alexperovich** [comment]
@dotnet-bot test Innerloop Linux ARM Emulator SoftFP Release Cross Build please

**Comment by yizhang82** [comment]
@AlexGhiondea FYI - is this duplicating your work?

**Comment by alexperovich** [comment]
@dotnet-bot help

**Comment by alexperovich** [comment]
@dotnet-bot test this please

**Pull request merged**